autofaiss/external/build.py

```diff
@@ -181,8 +181,11 @@ def create_index(
         del train_vectors
     else:
         train_size = 0
+
+    size_per_index = metadata.estimated_index_size_in_bytes() / nb_indices_to_keep
+
     memory_available_for_adding = cast_bytes_to_memory_string(
-        cast_memory_to_bytes(current_memory_available) - metadata.compute_memory_necessary_for_training(train_size)
+        cast_memory_to_bytes(current_memory_available) - size_per_index
```

Comment:
I don't think for the memory available for adding should be `total memory` - `size of index` because in `size_per_index` we have already considered the memory needed for adding all the vectors.
I am sure the code works but it will take smaller batch size.

Reply:
> because in size_per_index we have already considered the memory needed for adding all the vectors

yes that's exactly the memory that is needed at the end of adding

When you add vectors to the index, the index grows grows grows until its size is size_per_index
While doing that you also need to put in memory one batch

That's why the total size needed is memory_one_batch + size_per_index

so that makes the memory available for adding current memory_available - size_per_index

is that clearer?

Reply:
Yeah, it is clear. You are right.